src/Raven.Client/Json/Converters/JsonEnumerableConverter.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Concurrent;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+using Raven.Client.Documents.Conventions;
+using Sparrow.Json;
+
+namespace Raven.Client.Json.Converters
+{
+    internal class JsonEnumerableConverter : RavenJsonConverter
+    {
+        private readonly ConcurrentDictionary<Type, bool> _cache = new ConcurrentDictionary<Type, bool>();
+
+        private readonly DocumentConventions _conventions;
+
+        public override bool CanRead => false;
+
+        public JsonEnumerableConverter(DocumentConventions conventions)
+        {
+            _conventions = conventions ?? throw new ArgumentNullException(nameof(conventions));
+        }
+
+        public override bool CanConvert(Type objectType)
+        {
+            if (objectType == null)
+                return false;
+
+            return _cache.GetOrAdd(objectType, CanConvertInternal);
+        }
+
+        private bool CanConvertInternal(Type objectType)
+        {
+            if (objectType == typeof(LazyStringValue))
+                return false;
+
+            if (objectType == typeof(BlittableJsonReaderArray))
+                return false;
+
+            var jsonArrayContract = _conventions.JsonContractResolver.ResolveContract(objectType) as JsonArrayContract;
+            if (jsonArrayContract == null)
+                return false;
+
+            if (jsonArrayContract.IsMultidimensionalArray)
+                return false;
+
+            return jsonArrayContract.CollectionItemType != typeof(object);
```

Comment:
What is the type is an interface? `List<IFoo>` ?

Reply:
Need to double check, but shouldn't be an issue.

Reply:
Tested. It yields exactly the same JSON as before. Left code in the issue that I've used.